lib/index.js

```diff
@@ -39,6 +39,164 @@ Wigzo.prototype.loaded = function() {
 };
 
 /**
+ * Product Added
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAdded = function(track) {
+  var props = track.properties();
+  if (props.product_id) {
+    window.wigzo.track('addtocart', props.product_id);
+  }
+};
+
+
+/**
+ * Wishlist Product Added to Cart
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAddedFromWishlistToCart = function(track) {
+  var props = track.properties();
+  if (props.product_id) {
+    window.wigzo.track('addtocart', props.product_id);
+  }
+};
+
+/**
+ * Product Removed
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productRemoved = function(track) {
+  var props = track.properties();
+  if (props.product_id) {
+    window.wigzo.track('removedfromcart', props.product_id);
+  }
+};
+
+/**
+ * Products Searched
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productsSearched = function(track) {
+  var props = track.properties();
+  if (props.query) {
+    window.wigzo.track('search', props.query);
+  }
+};
+
+/**
+ * Product Added to Wishlist
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAddedToWishlist = function(track) {
+  var props = track.properties();
+  if (props.product_id) {
+    window.wigzo.track('wishlist', props.product_id);
+  }
+};
+
+
+/**
+ * Checkout Started
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.checkoutStarted = function(track) {
+  var productList = track.products();
+  var addedProductIds = [];
+  for (var i in productList) {
+    if (productList.hasOwnProperty(i)) {
+      addedProductIds.push(productList[i].product_id);
+    }
+  }
+
+  if (addedProductIds.length > 0) {
+    window.wigzo.track('checkoutstarted', addedProductIds);
+  }
+};
+
+/**
+ * Completed Order
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.orderCompleted = function(track) {
+  var productList = track.products();
+  var addedProductIds = [];
+  for (var i in productList) {
+    if (productList.hasOwnProperty(i)) {
+      addedProductIds.push(productList[i].product_id);
+    }
+  }
+
+  if (addedProductIds.length > 0) {
+    window.wigzo.track('buy', addedProductIds);
+  }
+};
+
+/**
+ * Product Review
+ *
+ * @param {Track} track
+ */
+Wigzo.prototype.productReviewed = function(track) {
+  window.wigzo.track('review', track.properties());
+};
+
+/**
+ * Product Clicked
+ *
+ * @param {Track} track
+ */
+Wigzo.prototype.productClicked = function(track) {
+  var props = track.properties();
+  var traits = reject({
+    productId : props.product_id,
+    title: props.name,
+    price: props.currency + ' ' + props.price,
+    category: props.category,
+    /* custom props */
+    image: props.imageUrl,
```

Comment:
`image`, `canonicalUrl`, `description` and `language` are all unspec'd properties of this event so if you really need this then you need to have folks send them via integration options:

analytics.track('Product Clicked', { normal props }, { Wigzo: {
    image: 'foo',
    canonicalUrl: 'bar',
    description: 'description',
    langauge: 'english'
}});